doc/reference/trd-syscalls.md

```diff
@@ -260,24 +260,51 @@ call to invoke. The kernel invokes callbacks only in response to Yield
 system calls.  This form of very limited preemption allows userspace
 to manage concurrent access to its variables.
 
-There are two Yield system calls:
+There are three Yield system calls:
   - `yield-wait`
   - `yield-no-wait`
+  - `yield-no-return`
 
 The first call, `yield-wait`, blocks until a callback executes. This is the
 only blocking system call in Tock. It is commonly used to provide a blocking
 I/O interface to userspace. A userspace library starts a long-running operation
 that has a callback, then calls `yield-wait` to wait for a callback. When the
 `yield-wait` returns, the process checks if the resuming callback was the one
-it was expecting, and if not calls `yield-wait` again. The `yield-wait` system
-call always returns `Success`.
+it was expecting, and if not calls `yield-wait` again. 
 
-The second call, 'yield-no-wait', executes a single callback if any is pending.
-If no callbacks are pending it returns immediately. The `yield-no-wait` system
-call returns `Success` if a callback executed and `Failure` if no callback
-executed.
+The second call, `yield-no-wait`, executes a single callback if any is pending.
+If no callbacks are pending it returns immediately. 
 
-The return values for Yield system calls are `Success` and `Failure`. 
+The third call, `yield-no-return` never returns and executes no callbacks. It
+causes the process to terminate. The kernel can reclaim any resources held by
+a process after it invokes `yield-no-return`.
```

Comment:
The name `yield-no-return` doesn't suggest to me that the process can end and be reclaimed. Is there a reason you picked `yield-no-return` over `yield-exit`?

Reply:
@jrvanwhy @phil-levis Yes, I think `yield-exit` makes it more clear that this system call would invoke the kernel to terminate the app and reclaim the app memory.  `yield-no-return` seems a bit more ambiguous.